### PR TITLE
EMCS-820 - Add correlationID in header and display it in Kibana

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/CorrelationIdService.scala [hunk 1]
```diff
@@ -16,11 +16,30 @@
...
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.http.HeaderCarrierConverter
+
 import java.util.UUID
 import javax.inject.{Inject, Singleton}
 
 @Singleton
 class CorrelationIdService @Inject() {
 
   def generateCorrelationId(): String = UUID.randomUUID().toString
```
Comment (oliverslade) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/CorrelationIdService.scala:29]: this either doesnt need brackets on it or the call on line 36 needs brackets adding

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/AdminDetailsController.scala [hunk 2]
```diff
@@ -31,7 +32,8 @@ import scala.concurrent.ExecutionContext
 class AdminDetailsController @Inject() (
   cc: ControllerComponents,
   auth: BackendAuthComponents,
-  movementRepository: MovementRepository
+  movementRepository: MovementRepository,
+  correlationIdService: CorrelationIdService
```
Comment (oliverslade) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/AdminDetailsController.scala:36]: looks like this is injected but never used?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/PreValidateTraderController.scala [hunk 3]
```diff
@@ -36,7 +37,8 @@ class PreValidateTraderController @Inject() (
   parseJsonAction: ParseJsonAction,
   preValidateTraderService: PreValidateTraderService,
   cc: ControllerComponents,
-  appConfig: AppConfig
+  appConfig: AppConfig,
+  correlationIdService: CorrelationIdService
```
Comment (oliverslade) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/PreValidateTraderController.scala:41]: In the PreValidateTraderConnector it still generates a new correlationId, is that correct?

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/SubmissionMessageService.scala [hunk 4]
```diff
@@ -43,8 +43,11 @@ class SubmissionMessageServiceImpl @Inject() (
     authorisedErn: String
   )(implicit hc: HeaderCarrier): Future[Either[EISErrorResponseDetails, EISSubmissionResponse]] = {
 
-    val correlationId = correlationIdService.generateCorrelationId()
-
+    val correlationId = hc
+      .headers(Seq(HttpHeader.xCorrelationId))
+      .find(_._1 == HttpHeader.xCorrelationId)
+      .map(_._2)
+      .getOrElse(correlationIdService.generateCorrelationId())
```
Comment (oliverslade) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/SubmissionMessageService.scala:50]: line break between these code blocks might help with readability

**Review by oliverslade — CHANGES_REQUESTED** [review]

**Review by Froom2 — APPROVED** [review]

**Review by oliverslade — APPROVED** [review]

**Pull request merged**